.expeditor/scripts/release_habitat/package_and_upload_binary.sh

```diff
@@ -7,128 +7,86 @@ set -euo pipefail
 
 source .expeditor/scripts/release_habitat/shared.sh
 
-export HAB_AUTH_TOKEN="${ACCEPTANCE_HAB_AUTH_TOKEN}"
 export HAB_BLDR_URL="${ACCEPTANCE_HAB_BLDR_URL}"
 
 channel=$(get_release_channel)
 
 echo "--- Channel: $channel - bldr url: $HAB_BLDR_URL"
 
-release_version=$(get_latest_pkg_release_version_in_release_channel "hab")
-
 declare -g hab_binary
 install_release_channel_hab_binary "x86_64-linux"
 
-${hab_binary} pkg install core/gzip \
-  core/tar \
-  core/wget \
-  core/zip
-
-# Import packages@chef.io GPG signing key
-aws s3 cp s3://chef-cd-citadel/packages_at_chef.io.pgp \
-  packages_at_chef.io.pgp \
-  --profile=chef-cd 
-gpg --import packages_at_chef.io.pgp
-
-tmp_root="$(mktemp -d -t "grant-XXXX")"
-extract_dir="$tmp_root/extract"
-mkdir -p "$extract_dir"
-
-echo "--- Downloading $release_version for $BUILD_PKG_TARGET"
-if [[ $BUILD_PKG_TARGET == *"darwin" ]]; then
-  artifact_name=$(buildkite-agent meta-data get MACOS_ARTIFACT)
-  buildkite-agent artifact download "$artifact_name" "$tmp_root/" --step "[:macos: build hab]"
-  mv "$tmp_root/$artifact_name" "$tmp_root/hab-$channel.hart"
-else
-  # once https://github.com/habitat-sh/habitat/issues/6878 is released,
-  # we can use hab pkg download rather than this wget
-  hab pkg exec core/wget wget \
-    "${HAB_BLDR_URL}/v1/depot/pkgs/core/hab/$release_version/download?target=$BUILD_PKG_TARGET" \
-    -O "$tmp_root/hab-$channel.hart"
-fi
+import_gpg_keys
+
+# Unsure we *really* need to bother with this tmp_root business, but
+# it does help to contain things a bit.
+tmp_root="$(mktemp -d -t "repackage-XXXX")"
+cd "${tmp_root}"
+
+echo "--- Downloading core/hab for $BUILD_PKG_TARGET from ${channel} channel"
+
+${hab_binary} pkg download core/hab \
+              --target="${BUILD_PKG_TARGET}" \
+              --channel="${channel}" \
+              --download-directory="."
+hart="$(find . -type f -name 'core-hab-*.hart')"
```

Comment:
*nit* May be safer to use `core-hab-*-$BUILD_PKG_TARGET.hart`?  This should always run on a clean machine and `$tmp_root` should be unique enough that the scenario where you'd end up with the wrong .hart is infinitesimally small.

Reply:
Hrmm... I could have sworn I had that in an earlier version of this code. I'll add that back, just because I like to be cautious.

Reply:
Added in https://github.com/habitat-sh/habitat/pull/7043/commits/87d51e601afc758bd105751f18e165bdb0a79b6e